classes/module/Module.php

```diff
@@ -3432,33 +3438,54 @@ public function isAdminLegacyContext()
      */
     public function isSymfonyContext()
     {
-        return !$this->isAdminLegacyContext() && defined('_PS_ADMIN_DIR_');
+        // Check that a container is available but not generated by PrestaShop
+        $container = $this->getContainer();
+
+        return null !== $container && !($container instanceof LegacyContainerInterface);
```

Comment:
I think we should keep this as is. It's the uses that must be updated to perform duck-typing (aka "is the container present?")

Reply:
But having a container doesn't tell if it's a full filled container from Symfony (with CQRS, bundles, and so on)
Or a legacy one with restricted services and features

So we could leave `isSymfonyContext` untouched, but add a new method to easily test which kind of container is present

Although I think the behaviour of this method has already been modified since 175, because some native modules like ps_linklist or productcomments can't be installed via CLI any more, so something changed

Reply:
@eternoendless I reverted the changes on `isSymfonyContext` so no more changes nor BC break
And instead I added two methods `hasSymfonyContainer|hasLegacyContainer` which will allow to detect the context/container more easily

Reply:
I don't think we should add more "maybe useful someday" methods in a class which already does too much.

Reply:
Ok I'll remove this one, and explain this in the PHPDoc of `getContainer`